### PR TITLE
feat(staking): implement claim rewards panel + dialog

### PR DESCRIPTION
**Pull request by moo-onthelawn**
Screenshot 2024-06-05 at 4 31 14 PM
Screenshot 2024-06-05 at 4 31 10 PM

---

## Views

* New: `StakingRewardDialog`
  * ref [design](https://www.figma.com/design/dhSd8kkcUrY3UBRsdS89kH/dYdX-%E2%80%BA-Legal?node-id=912-6177&t=X7Dc8EtrwrbS3ryg-0)
  * We made the decision to cut buying power from MVP; also made decision that gas should be paid in USDC
  * Gas fee is currently hardcoded, TODO to implement simulation for actual gas fee
  * Clicking claim rewards currently does nothing, TODO to implement it

* `RewardsPage`
  * Render `StakingRewardPanel` when flag enabled && there is a non zero amount of reward to claim
  * Read in staking rewards (from keplr) using `getStakingRewards` selector; then reading from `totalRewards`
    * I think there should technically only ever be one entry where the `denom === usdcDenom` but summing it up via reduce seems more accurate (since we're returned an array) 
  * Currently a bug in the way we get back the rewards (in parsing the Dec); so I'm manually dividing by 10**24 though it should be 6 🤔 

## Components

* `Dialog.tsx`
  * Optionally allow callers to disable blur on header (necessary for `StakingRewardDialog`)

* `StakingRewardPanel.tsx`
  * Pretty self explanatory; I think the mocks only show up to 2 decimal points; I have it at 4 right now because my current testing account has so little rewards that I keep seeing 0 if I keep it to 2. I wonder if we want to do something similar to Keplr where if it's <0.01 USDC, we just show <0.01?

* `Panel.tsx`
  * Updated `border-radius` to point to variable so that callers can share it

## Styles/Mixins

* `styles/text.css`
  * Added --font-extra-bold for use in  `StakingRewardDialog`

* `styles/layoutMixins.ts`
  * Updated `backdrop-filter` to point to variable (so that we can disable it in `StakingRewardDialog`)
  
* `styles/globalStyle.ts`
  * Add `--color-green-dark` (used for background of `StakingRewardDialog` and `StakingRewardPanel`)

* `styles/themes.ts`, `styles/colors.ts`, `styles/base.ts`
  * Add `greenDark`, using the same for light/dark/classic right now, can follow up with Design on variants
  - [ ] TODO: follow up with design

## Testing
- `tradingrewardsrehaul=true&staking=true`
  - next PR will merge the two flags, I just didn't wanna rebase this PR yet
---

### CODE REVIEW

**Review by yogurtandjam — APPROVED** [review]

**Comment by yogurtandjam** [comment]
ty for todo comments 🙏

**Review by rosepuppy — APPROVED** [review]
looks great!!

File: src/pages/token/rewards/RewardsPage.tsx [hunk 1]
```diff
@@ -36,13 +44,30 @@ const RewardsPage = () => {
...
+  }, 0);
+
   const showMigratePanel = import.meta.env.VITE_V3_TOKEN_ADDRESS && isNotTablet;
   const showGeoblockedPanel =
     tradingRewardsRehaulEnabled && complianceState !== ComplianceStates.FULL_ACCESS;
+  const showStakingRewardPanel = totalUsdcRewards > 0 && !showGeoblockedPanel && enableStaking;
 
+  const stakingRewardPanel = (
+    <StakingRewardPanel
+      usdcRewards={MustBigNumber(totalUsdcRewards).div(Number(`1e${usdcDecimals}`))}
```
Comment (rosepuppy) [inline src/pages/token/rewards/RewardsPage.tsx:68]: nit: you can use `formatUnits` to do this.

Comment (moo-onthelawn) [reply src/pages/token/rewards/RewardsPage.tsx:68]: ooh ok will follow up with this in my merge all flags pr

**Pull request merged**